code/modules/mob/living/simple_animal/hostile/hostile.dm

```diff
@@ -573,15 +580,17 @@
 
 ////// AI Status ///////
 /mob/living/simple_animal/hostile/proc/AICanContinue(list/possible_targets)
+	if(QDELETED(src))
+		return FALSE
 	switch(AIStatus)
 		if(AI_ON)
-			. = 1
+			. = TRUE
 		if(AI_IDLE)
 			if(FindTarget(possible_targets))
-				. = 1
+				. = TRUE
 				toggle_ai(AI_ON) //Wake up for more than one Life() cycle.
```

Comment:
```suggestion
				toggle_ai(AI_ON) //Wake up for more than one Life() cycle.
				return TRUE
```

may as well fix this while we're in the area. if we runtime we really shouldn't be returning a "positive" result anyways